### PR TITLE
[WIP] Cleanup the way boost is located/used in the cmake files

### PR DESCRIPTION
**Pull request by greglandrum**
Boost now (since v1.70) has its own cmake integration which theoretically should make things simpler/easier. This PR explores using that.
It looks like we need boost 1.73 or later to make the python integration work.

This PR simplifies the cmake files a bit and removes the need to include `-DBoost_NO_BOOST_CMAKE=True` in the cmake invocation. 

Other changes here:
1. bump minimum boost version to 1.73 (released: April 2020)
2. bump minimum cmake version to 3.8 (not sure this is necessary, released early 2017)
3. The only code actual changes are adding some forgotten EXPORTS in the MolDraw2D code base; otherwise this is all changes to cmake files and the configuration of the CI builds.

### CODE REVIEW

**Comment by greglandrum** [comment]
@bp-kelley @ptosco @ricrogz : what do you think about adding the boost >=1.70 dependency? 
Do you see any problems with that?

**Comment by ptosco** [comment]
@greglandrum No, that’s fine for me.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7)
+cmake_minimum_required(VERSION 3.8)
```
Comment (ricrogz) [inline CMakeLists.txt:1]: I think this version of CMake does not support Boost 1.70.

Somewhen I had that problem, I think with a recent boost version (I might have built it from source myself) on a not so recent version of Ubuntu, and somehow I ended in this thread that mentions which CMake versions support which Boost versions (rule of thumb is always use a CMake version which was released later than the Boost libs version you are using):

https://stackoverflow.com/questions/42123509/cmake-finds-boost-but-the-imported-targets-not-available-for-boost-version

According to that, we should bump this to 3.14 to support Boost 1.70.

**Review by ricrogz — COMMENTED** [review]
Sounds good to me, although we should require a higher version of CMake (see my comment below).

File: CMakeLists.txt (hunk 1 above)
Comment (greglandrum) [reply CMakeLists.txt:1]: I believe this is only true if you aren’t using the boost cmake integration (i.e. the way we are currently doing it), but I will confirm that this fork actually builds with cmake 3.8

Comment (ricrogz) [reply CMakeLists.txt:1]: Ok, that makes sense to me.

**Comment by bp-kelley** [comment]
> @bp-kelley @ptosco @ricrogz : what do you think about adding the boost >=1.70 dependency?
> Do you see any problems with that?

Nope.  Moving forwards is always good in my opinion.

File: CMakeLists.txt (hunk 1 above)
Comment (greglandrum) [reply CMakeLists.txt:1]: That took longer to finish than I had hoped.
I found some indication, though now I can't remember what it was, that we require cmake 3.10 on at least some systems. Of course, I just did a successful build with 3.8 on linux.

What is a larger concern, and I will update the PR description for this, is that it seems like we need boost>=1.73. I tracked this down using the conda-forge boost builds and need to confirm with my own boost builds.

**Pull request closed without merging**